full-service/src/service/wallet_api.rs

```diff
@@ -0,0 +1,1119 @@
+    //         ledger_db,
+    //     )
+    // }
+
+    fn dispatch(client: &Client, body: JsonValue, logger: &Logger) -> serde_json::Value {
+        let mut res = client
+            .post("/wallet")
+            .header(ContentType::JSON)
+            .body(body.to_string())
+            .dispatch();
+        assert_eq!(res.status(), Status::Ok);
+        let body = res.body().unwrap().into_string().unwrap();
+        log::info!(logger, "Attempted dispatch got response {:?}", body);
+
+        let res: JsonValue = serde_json::from_str(&body).unwrap();
+        res.get("result").unwrap().clone()
+    }
+
+    fn dispatch_expect_error(
+        client: &Client,
+        body: JsonValue,
+        logger: &Logger,
+        expected_err: String,
+    ) {
+        let mut res = client
+            .post("/wallet")
+            .header(ContentType::JSON)
+            .body(body.to_string())
+            .dispatch();
+        assert_eq!(res.status(), Status::Ok);
+        let body = res.body().unwrap().into_string().unwrap();
+        log::info!(logger, "Attempted dispatch got response {:?}", body);
+        assert_eq!(body, expected_err);
+    }
+
+    #[test_with_logger]
+    // A create_account RPC should correctly invoke the wallet service's create_account method.
+    fn test_create_account(logger: Logger) {
+        let rocket_config = rocket::Config::build(rocket::config::Environment::Development).port(get_free_port()).unwrap();
+
+        // Create Account request body.
+        let body = json!({
+            "method": "create_account",
+            "params": {
+                "name": "Alice Main Account",
+            }
+        });
+
+        let expected_response: JsonCreateAccountResponse = {
+            let mut account = JsonAccount::default();
+            account.account_id = "123".to_string();
+            JsonCreateAccountResponse{ entropy: "ABC".to_string(), account}
+        };
+
+        let mut mock_wallet_service = MockWallet::new();
+        {
+            let expected = expected_response.clone();
+            // create_account should be called exactly once.
+            mock_wallet_service.expect_create_account()
+                .return_once(move |_name, _first_block| Ok(expected));
+        }
+
+        let state = WalletApiState{service: Arc::new(Mutex::new(mock_wallet_service))};
+        let rocket = rocket(rocket_config, state);
+        let client = Client::new(rocket).unwrap();
+
+        // The result should be the expected response.
+        let result = dispatch(&client, body, &logger);
+        assert!(result.get("entropy").is_some());
+        assert_eq!(result.get("entropy").unwrap(), expected_response.entropy.as_str());
+
+        let account_obj = result.get("account").unwrap();
+        assert_eq!(account_obj.get("account_id").unwrap(), expected_response.account.account_id.as_str());
+    }
+
+    // #[test_with_logger]
+    // fn test_account_crud(logger: Logger) {
```

Comment:
These unit tests seem to focus on checking the implementation of the WalletService, and so the "meat" of these tests should probably get moved to wallet_service.rs